modules/indexer/code/wrapped.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package code
+
+import (
+	"fmt"
+	"sync"
+)
+
+var (
+	indexer = newWrappedIndexer()
```

Comment:
The indexer has to be wrapped to allow searches etc to hold whilst we are waiting for the indexer to boot up further we must not run functions on the indexer until it is completely ready.

If the indexer is closed before its internal element is present then we will return a simple error rather than panic.